pkg/controller/build/build_controller.go

```diff
@@ -312,48 +315,58 @@ func (r *ReconcileBuild) validateSecrets(ctx context.Context, secretNames []stri
 	return nil
 }
 
-func (r *ReconcileBuild) configFinalizer(ctx context.Context, b *build.Build) error {
-	if b.GetAnnotations()[build.AnnotationBuildRunDeletion] == "true" {
-		if !contains(b.GetFinalizers(), build.BuildFinalizer) {
-			ctxlog.Info(ctx, "add finalizer to build", namespace, b.Namespace, name, b.Name)
-			b.SetFinalizers(append(b.GetFinalizers(), build.BuildFinalizer))
-			if err := r.client.Update(ctx, b); err != nil {
-				return err
+// validateBuildRunOwnerReferences defines or removes the ownerReference for the BuildRun based on
+// an annotation value
+func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b *build.Build) error {
+	buildRunList, err := r.retrieveBuildRunsfromBuild(ctx, b)
+	if err != nil {
+		return err
+	}
+
+	if b.GetAnnotations()[build.AnnotationBuildRunDeletion] == "false" {
+		// if the buildRun have an ownerreference to the Build, lets remove it
+		for _, buildRun := range buildRunList.Items {
+			if index := r.validateBuildOwnerReference(buildRun.OwnerReferences, b); index != -1 {
+				buildRun.OwnerReferences = removeOwnerReferenceByIndex(buildRun.OwnerReferences, index)
+				if err := r.client.Update(ctx, &buildRun); err != nil {
+					return err
+				}
+				ctxlog.Info(ctx, fmt.Sprintf("succesfully updated BuildRun %s", buildRun.Name), namespace, buildRun.Namespace, name, buildRun.Name)
 			}
 		}
 	} else {
```

Comment:
if it is `else`, the end user may set the `AnnotationBuildRunDeletion` to `yes` or `aha`, then it will also enter this logic which is not good.

Reply:
Good point, I would suggest to rewrite the `if`/`else` to a `switch`:

```go
switch b.GetAnnotations()[build.AnnotationBuildRunDeletion] {
	case "true":
		// ...

	case "false":
		// ...

	default:
		// error case ...
	}
```

Could also improve readability.

Reply:
this was on purpose, the important value was "false", but I agree it should be more strict. Fixed.

Reply:
hold on, I didnt see @HeavyWombat comment, enhancing

Reply:
@HeavyWombat so I added this in https://github.com/shipwright-io/build/blob/98b81eba2346fc1dc7c9f58113e99a4abb741e12/pkg/controller/build/build_controller.go#L326-L358 

There is one caveat, if the user defines a wrong annotation value (_not true or false_), then the logic will return an error (_ignored in the reconcile parent func_) and we will get an INFO log (_ that illustrates the problem_). Ideally I think we want the `Build` to reflect this issue in its Status.Reason and fail the `Build` Registration. 

But I will prefer to do the above in a second pr, mainly because this use-case was not supported before. Opinions?

Reply:
Agree, an annotation value that is neither `true` nor `false` is good to be reflected in the build status.I am okay doing this separately.